src/shim/executor.rs

```diff
@@ -195,6 +211,144 @@ impl From<Receipt_v3> for Receipt {
     }
 }
 
+#[derive(Clone, Debug)]
+pub enum Entry {
+    V3(Entry_v3),
+    V4(Entry_v4),
+}
+
+impl From<Entry_v3> for Entry {
+    fn from(other: Entry_v3) -> Self {
+        Self::V3(other)
+    }
+}
+
+impl From<Entry_v4> for Entry {
+    fn from(other: Entry_v4) -> Self {
+        Self::V4(other)
+    }
+}
+
+impl Entry {
+    #[cfg(test)]
+    pub fn new(
+        flags: crate::shim::fvm_shared_latest::event::Flags,
+        key: String,
+        codec: u64,
+        value: Vec<u8>,
+    ) -> Self {
+        Entry::V4(Entry_v4 {
+            flags,
+            key,
+            codec,
+            value,
+        })
+    }
+
+    pub fn into_parts(self) -> (u64, String, u64, Vec<u8>) {
+        match self {
+            Self::V3(v3) => {
+                let Entry_v3 {
+                    flags,
+                    key,
+                    codec,
+                    value,
+                } = v3;
+                (flags.bits(), key, codec, value)
+            }
+            Self::V4(v4) => {
+                let Entry_v4 {
+                    flags,
+                    key,
+                    codec,
+                    value,
+                } = v4;
+                (flags.bits(), key, codec, value)
+            }
+        }
+    }
+
+    pub fn value(&self) -> &Vec<u8> {
+        match self {
+            Self::V3(v3) => &v3.value,
+            Self::V4(v4) => &v4.value,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum ActorEvent {
+    V3(ActorEvent_v3),
+    V4(ActorEvent_v4),
+}
+
+impl From<ActorEvent_v3> for ActorEvent {
+    fn from(other: ActorEvent_v3) -> Self {
+        ActorEvent::V3(other)
+    }
+}
+
+impl From<ActorEvent_v4> for ActorEvent {
+    fn from(other: ActorEvent_v4) -> Self {
+        ActorEvent::V4(other)
+    }
+}
+
+impl ActorEvent {
+    pub fn entries(&self) -> Vec<Entry> {
+        match self {
+            Self::V3(v3) => v3.entries.clone().into_iter().map(Into::into).collect(),
+            Self::V4(v4) => v4.entries.clone().into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum StampedEvent {
+    V3(StampedEvent_v3),
+    V4(StampedEvent_v4),
+}
+
+impl From<StampedEvent_v3> for StampedEvent {
+    fn from(other: StampedEvent_v3) -> Self {
+        StampedEvent::V3(other)
+    }
+}
+
+impl From<StampedEvent_v4> for StampedEvent {
+    fn from(other: StampedEvent_v4) -> Self {
+        StampedEvent::V4(other)
+    }
+}
+
+impl StampedEvent {
+    #[allow(dead_code)]
+    pub fn get_events(db: &impl Blockstore, events_cid: &Cid) -> anyhow::Result<Vec<StampedEvent>> {
+        let mut events = Vec::new();
+        if let Ok(amt) = Amt::<StampedEvent_v4, _>::load(events_cid, db) {
+            amt.for_each(|_, event| {
+                events.push(event.clone().into());
+                Ok(())
+            })?;
```

Comment:
Isn't `for_each` deprecated in favour of `iter()`?

https://github.com/filecoin-project/ref-fvm/blob/488f704015a83de0db1737f5f47305939ea74f86/ipld/amt/src/amt.rs#L451-L462

Reply:
Good catch.

Reply:
Let's just remove this. It's deadcode anyway.